pkg/apis/duck/v1beta1/subscribable_types_conversion.go

```diff
@@ -75,41 +85,54 @@ func (source *SubscribableStatus) ConvertTo(ctx context.Context, sink *eventingd
 			}
 		}
 	}
+	return nil
 }
 
 // ConvertFrom implements apis.Convertible.
 func (sink *Subscribable) ConvertFrom(ctx context.Context, from apis.Convertible) error {
 	switch source := from.(type) {
 	case *eventingduckv1.Subscribable:
 		sink.ObjectMeta = source.ObjectMeta
-		sink.Status.ConvertFrom(ctx, source.Status)
-		sink.Spec.ConvertFrom(ctx, source.Spec)
+		if err := sink.Status.ConvertFrom(ctx, source.Status); err != nil {
+			return err
+		}
+		if err := sink.Spec.ConvertFrom(ctx, source.Spec); err != nil {
+			return err
+		}
 		return nil
 	default:
 		return fmt.Errorf("unknown version, got: %T", source)
 	}
 }
 
 // ConvertFrom helps implement apis.Convertible
-func (sink *SubscribableSpec) ConvertFrom(ctx context.Context, source eventingduckv1.SubscribableSpec) {
+func (sink *SubscribableSpec) ConvertFrom(ctx context.Context, source eventingduckv1.SubscribableSpec) error {
 	if len(source.Subscribers) > 0 {
 		sink.Subscribers = make([]SubscriberSpec, len(source.Subscribers))
 		for i, s := range source.Subscribers {
-			sink.Subscribers[i].ConvertFrom(ctx, s)
+			if err := sink.Subscribers[i].ConvertFrom(ctx, s); err != nil {
+				return err
+			}
 		}
 	}
+	return nil
 }
 
 // ConvertFrom helps implement apis.Convertible
-func (sink *SubscriberSpec) ConvertFrom(ctx context.Context, source eventingduckv1.SubscriberSpec) {
+func (sink *SubscriberSpec) ConvertFrom(ctx context.Context, source eventingduckv1.SubscriberSpec) error {
 	sink.UID = source.UID
 	sink.Generation = source.Generation
 	sink.SubscriberURI = source.SubscriberURI
 	sink.ReplyURI = source.ReplyURI
 	sink.Delivery = &DeliverySpec{
 		DeadLetterSink: &duckv1.Destination{},
 	}
-	sink.Delivery.ConvertFrom(ctx, source.Delivery)
+	if source.Delivery != nil {
+		if err := sink.Delivery.ConvertFrom(ctx, source.Delivery); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
There was a segfault here.

All the other changes in the PR is about proper error handling.
@vaikas if it is too much changes, feel free to pick the changes you like